app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -220,12 +229,46 @@ class PipelineViz extends React.Component {
     return stepClassName.replace(/^(PipelineStep(Run|Generate)?)/, "");
   }
 
-  handleStepClick(stageIndex, info) {
-    const clickedNodeId = info.nodes[0];
+  inverseTransformDOMCoordinates(x, y) {
```

Comment:
Can you do a small refactor here? Move `PipelineViz.jsx` and `pipeline_vis.scss` to a folder named `PipelineViz` and move this generic transform conversion functions to a util file within that folder. 
Just trying to hide some of this complexity, since is not a core component and should not be a an instance method. It can be a generic function that receives a container and transforms the coordinates.